modules/api/php/provisioners/projectdicomsrowprovisioner.class.inc

```diff
@@ -0,0 +1,103 @@
+<?php declare(strict_types=1);
+/**
+ * This file implements a data provisioner to get all dicoms of a project
+ * created since a given date.
+ *
+ * PHP Version 7
+ *
+ * @category API
+ * @package  Loris
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\api\Provisioners;
+
+use \LORIS\Data\Provisioners\DBRowProvisioner;
+use \LORIS\api\Models\ProjectDicomsRow;
+/**
+ * This file implements a data provisioner to get all dicoms of a project
+ * created since a given date.
+ *
+ * PHP Version 7
+ *
+ * @category API
+ * @package  Loris
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class ProjectDicomsRowProvisioner extends DBRowProvisioner
+{
+    /**
+     * Create a RowProvisioner
+     *
+     * @param \Project  $project The project from which dicoms are requested
+     * @param \DateTime $since   The project from which dicoms are requested
+     */
+    function __construct(\Project $project, \DateTime $since)
+    {
+        parent::__construct(
+            '
+             SELECT
+               s.CandID as Candidate,
+               c.PSCID as PSCID,
+               c.Entity_type as Entity_type,
+               s.Visit_label as Visit,
+               s.Date_visit as Visit_date,
+               s.CenterID as CenterID,
+               p.Name as Site,
+               f.InsertTime as InsertTime,
+               t.TarchiveID as tarchiveid,
+               t.DicomArchiveID as DicomArchiveID,
+               t.ArchiveLocation as Archive,
+	       t.SourceLocation as Source,
+               f.TarchiveSource as TarchiveSource,
+               f.AcquisitionProtocolID as AcquisitionProtocolID,
+               f.ProcessProtocolID as ProcessProtocolID,
+               ts.TarchiveSeriesID as TarchiveSeriesID
+             FROM
+               files f
+             LEFT JOIN session s
+               ON (f.SessionID = s.ID)
+             LEFT JOIN candidate c
+               ON (s.CandID = c.CandID)
+             LEFT JOIN psc p
+               ON (s.CenterID = p.CenterID)
+             LEFT JOIN Project project
+               ON (c.RegistrationProjectID = project.ProjectID)
+	     LEFT JOIN tarchive t
+               ON (f.TarchiveSource = t.TarchiveID)
+             LEFT JOIN tarchive_series ts
+               ON (ts.TarchiveID = t.TarchiveID)
+	     LEFT JOIN server_processes sp
+               ON (sp.pid = f.ProcessProtocolID)
+             WHERE
+               c.Active = \'Y\' AND
+               s.Active = \'Y\' AND
+               project.Name = :v_projectname AND
+               f.InsertTime > :v_time
+             ORDER BY f.InsertTime ASC
```

Comment:
I would simplify the query for this endpoint. I guess the goal of the endpoint is to have a DICOMs project endpoint that lists all the tarchive files in the database. In that case, there is no need to join to the files table or the tarchive_series table. Below is the simplification I was thinking about.

```suggestion
             SELECT
               s.CandID as Candidate,
               c.PSCID as PSCID,
               c.Entity_type as Entity_type,
               s.Visit_label as Visit,
               s.Date_visit as Visit_date,
               s.CenterID as CenterID,
               p.Name as Site,
               t.DateAcquired as date_acquired,
               t.DateFirstArchived as date_first_archived,
               t.DateLastArchived as date_last_archived,
               t.TarchiveID as tarchiveid,
               t.DicomArchiveID as DicomArchiveID,
               t.ArchiveLocation as Archive,
	           t.SourceLocation as Source,
             FROM
               tarchive t
             LEFT JOIN  session s
               ON (t.SessionID = s.ID)
             LEFT JOIN candidate c
               ON (s.CandID = c.CandID)
             LEFT JOIN psc p
               ON (s.CenterID = p.CenterID)
             LEFT JOIN Project project
               ON (c.RegistrationProjectID = project.ProjectID)
             WHERE
               c.Active = \'Y\' AND
               s.Active = \'Y\' AND
               project.Name = :v_projectname AND
               t.DateLastArchived > :v_time
             ORDER BY t.DateLastArchived ASC
```

Reply:
yes, that is the idea. I'll have a look at this, thank you for the suggestion!